specification/testbase/resource-manager/Microsoft.TestBase/preview/2023-11-01-preview/draftpackage.json

```diff
@@ -0,0 +1,1142 @@
+              "$ref": "#/definitions/CopyFromPackageOperationParameters"
+            },
+            "description": "Parameters supplied to the Test Base Draft Package CopyFromPackage operation."
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The operation was accepted. Follow the Location header to poll for final outcome.",
+            "headers": {
+              "Azure-AsyncOperation": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              },
+              "Location": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DraftPackageCopyFromPackage": {
+            "$ref": "./examples/DraftPackageCopyFromPackage.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.TestBase/testBaseAccounts/{testBaseAccountName}/draftPackages/{draftPackageName}/extractFile": {
+      "post": {
+        "tags": [
+          "Test Base Draft Package"
+        ],
+        "operationId": "DraftPackages_ExtractFile",
+        "description": "Performs extracting file operation for a Test Base Draft Package",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./testbase.json#/parameters/TestBaseAccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DraftPackageNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ExtractFileOperationParameters"
+            },
+            "description": "Parameters supplied to the Test Base Draft Package ExtractFile operation."
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The 'ExtractFile' operation of Draft Package was accepted. Follow the Location header to poll for final outcome.",
```

Comment:
>description [](http://example.com/codeflow?start=13&length=11)

you still need a 200 or 204 final response code, in case they use the 202 Location header polling pattern

Reply:
According to https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-response-codes.md, as this operation doesn't have response payload, only 202 is allowed. The effect would be take on the draft package resource model itself. The location header will provide the operationstatus endpoint instead of this api.